arangod/IResearch/IResearchAqlAnalyzer.cpp

```diff
@@ -486,13 +527,63 @@ bool AqlAnalyzer::next() {
       while (_queryResults->numRows() > _resultRowIdx) {
         AqlValue const& value =
             _queryResults->getValueReference(_resultRowIdx++, _engine.resultRegister());
-        if (value.isString() || (value.isNull(true) && _options.keepNull)) {
-          if (value.isString()) {
-            _term.value = arangodb::iresearch::getBytesRef(value.slice());
-          } else {
-            _term.value = irs::bytes_ref::EMPTY;
+        if (value.isString() ||
+            (_options.keepNull && value.isNull(true) ) ||
+            (_options.returnType != AnalyzerValueType::Undefined && !value.isNull(true))) {
+          switch (_options.returnType) {
+            case AnalyzerValueType::Undefined:
```

Comment:
I suggest we get rid of  `AnalyzerValueType::Undefined` as it only adds extra complexity without providing any benefits

Reply:
As we already released documentation I wanted to retain that behaviour: no implicit conversion (as setting returnType:'string' will enable conversion). But as we have not released final version of binaries we could drop support and declare "always implicit conversion to string by default". WDYT?